testdata/time_date_decimal_literal.go

```diff
@@ -0,0 +1,98 @@
+package pkg
+
+import "time"
+
+// these should match and be reported as invalid usage of time.Date
+// the rule should suggest to use a decimal number
+var (
+	// All these examples refers to the same date
+	// 2023-01-02 03:04:05.000000006 +0000 UTC
+
+	// this is an invalid usage
+	// people are using this format because a date is easier to read
+	// but here it leads to use octal numbers0
+	_ = time.Date(
+		2023,
+		01,        // MATCH /use decimal digits for time.Date month argument: octal notation found: use 1 instead of 01/
+		02,        // MATCH /use decimal digits for time.Date day argument: octal notation found: use 2 instead of 02/
+		03,        // MATCH /use decimal digits for time.Date hour argument: octal notation found: use 3 instead of 03/
+		04,        // MATCH /use decimal digits for time.Date minute argument: octal notation found: use 4 instead of 04/
+		05,        // MATCH /use decimal digits for time.Date second argument: octal notation found: use 5 instead of 05/
+		000000006, // MATCH /use decimal digits for time.Date nanosecond argument: octal notation found: use 6 instead of 000000006/
+		time.UTC)
+
+	// the following one could have been written by someone who is not aware of the issue
+	// Please note, there are multiple issues on the same line
+	//
+	// use special syntax to match multiple issues being reported on the same line
+	// MATCH:34 /use decimal digits for time.Date month argument: octal notation found: use 1 instead of 01/
+	// MATCH:34 /use decimal digits for time.Date day argument: octal notation found: use 2 instead of 02/
+	// MATCH:34 /use decimal digits for time.Date hour argument: octal notation found: use 3 instead of 03/
+	// MATCH:34 /use decimal digits for time.Date minute argument: octal notation found: use 4 instead of 04/
+	// MATCH:34 /use decimal digits for time.Date second argument: octal notation found: use 5 instead of 05/
+	// MATCH:34 /use decimal digits for time.Date nanosecond argument: octal notation found: use 6 instead of 000000006/
+	_ = time.Date(2023, 01, 02, 03, 04, 05, 000000006, time.UTC)
+)
+
+// gofumpt formats legacy non-decimal notation to new one non-decimal notation
+// it transforms 01, 02, 03, 04, 05, 06, and 07 to 0o1, 0o2, 0o3, 0o4, 0o5, 0o6, and 0o7
+// but here with time.Date it doesn't make sense. This is the main reason why the rule was created.
+var (
+	_ = time.Date(2023, 0o1, 2, 3, 4, 5, 6, time.UTC) // MATCH /use decimal digits for time.Date month argument: octal notation found: use 1 instead of 0o1/
```

Comment:
If a person deliberately set it to octal, then I think in this case it's a style issue, not a real problem. So, I'm not sure we should do it in this linter rule.

Reply:
as explained [here](https://github.com/mgechev/revive/issues/1326#issuecomment-2841292811), it would be reported but with low confidence